flink-clients/src/test/java/org/apache/flink/client/program/ClientTest.java

```diff
@@ -139,6 +141,69 @@ public void shouldSubmitToJobClient() {
 	}
 
 	/**
+	 * Tests that invalid detached mode programs fail.
+	 */
+	@Test
+	public void testDetachedMode() throws Exception{
+		jobManagerSystem.actorOf(Props.create(SuccessReturningActor.class), JobManager.JOB_MANAGER_NAME());
+		Client out = new Client(config);
+
+		try {
+			PackagedProgram prg = new PackagedProgram(TestExecuteTwice.class);
+			out.runDetached(prg, 1);
+		} catch (ProgramInvocationException e) {
```

Comment:
The problem with this test is that it doesn't fail if the Program actually runs (which is the aim of the test...).